### PR TITLE
Issue #627: WARNING Invalid JSON data from key

### PR DESCRIPTION
**Pull request by zaraG**
Couldn't reproduce any invalid JSON error, expect when there is no key found.

I catched this one and throw a 404 error.

### CODE REVIEW

File: src/leap/keymanager/__init__.py [hunk 1]
```diff
@@ -232,6 +234,40 @@ def _get_key_from_nicknym(self, address):
...
+    def _key_search_failed(self, code):
+        """
+        Checks if response code is NOT_FOUND.
+
+        :param code: The response code, after trying to fecth a key from nicknym.
+        :type code: int
+
+        :return: boolean
+        """
+        return code == NOT_FOUND
```
Comment (meskio) [inline src/leap/keymanager/__init__.py]: I don't think this method is needed. The docstring is way longer than the code. I think it's easier to read if this code lives directly in the if below.

**Comment by zaraG** [comment]
Ok, should I change it, or will you do that?

**Comment by meskio** [comment]
Can you squash the commits into one or two? we don't need the 'refactor' commits in the log. Also, we have a commit template with the format we like to see in the commit messages:
https://github.com/leapcode/keymanager/blob/develop/docs/leap-commit-template.README

**Comment by meskio** [comment]
> Ok, should I change it, or will you do that?

It's better if you update the pull-req yourself.

**Pull request merged**